cypress/integration/product/change_product_price_spec.js

```diff
@@ -0,0 +1,90 @@
+import { Product, ProductCategory, ProductPrice} from '../../support/utils/product';
+import { SalesOrder, SalesOrderLine} from '../../support/utils/sales_order';
+import { BPartner} from '../../support/utils/bpartner';
+import { BPartnerLocation } from '../../support/utils/bpartner_ui';
+
+describe('Create Product', function() {
+  const timestamp = new Date().getTime(); 
+  const productName = `ProductName ${timestamp}`;
+  const productValue = `ProductNameValue ${timestamp}`;
+  const productCategoryName = `ProductCategoryName ${timestamp}`;
+  const productCategoryValue = `ProductNameValue ${timestamp}`;
+  const bpName = `Customer ${timestamp}`;
+  let salesOrderRecordID;
+  let productRecordID;
+
+  it('Create a new ProductCategory', function() {
+    cy.fixture('product/simple_productCategory.json').then(productCategoryJson => {
+      Object.assign(new ProductCategory(), productCategoryJson)
+        .setName(productCategoryName)
+        .setValue(productCategoryValue)
+        .apply();
+    });
+  });
+
+  it('Create a Product and Price', function() {
+    const productPrice = new ProductPrice()
+      .setPriceList("Testpreise Kunden (Deutschland)_Germany - Deutschland_EUR_2015-01-01")
+      .setTaxCategory("Regular Tax Rate 19% (Germany)")
+      .setListPriceAmount(`0`)
+      .setStandardPriceAmount(`0`)
+      .setLimitPriceAmount(`0`);
+
+    cy.fixture('product/simple_product.json').then(productJson => {
+      Object.assign(new Product(), productJson)
+        .setName(productName)
+        .setValue(productValue)
+        .setProductCategory(productCategoryValue + '_' + productCategoryName)
+        .addProductPrice(productPrice)
+        .setSold(false)
+        .apply();
+    });
+
+    cy.getCurrentWindowRecordId().then(recordId => {
+      productRecordID = recordId;
+    });
+  });
+
+  it('create BP', function() {
+    new BPartner({ name: bpName })
+      .setCustomer(true)
+      .setPaymentTerm('Immediatlely')
+      .setCustomerPricingSystem(`Testpreisliste Kunden`)
+      .addLocation(new BPartnerLocation('Address1').setCity('Cologne').setCountry('Deutschland'))
+      .apply();
+  });
+
+  it('Create a Sales Order', function() {
+    const salesOrderLine = new SalesOrderLine()
+      .setProduct(productName)
+      .setQuantity(`1`);
+
+    cy.fixture('sales/sales_order.json').then(salesOrderJson => {
+      Object.assign(new SalesOrder(), salesOrderJson)
+      .setBPartner(bpName)
+      .addLine(salesOrderLine)
+      .apply();
+    });
+
+    cy.getCurrentWindowRecordId().then(recordId => {
+      salesOrderRecordID = recordId;
+    });
+  });
+
+  it(`Change Price and use it`, function(){
+    cy.visitWindow(`140`, productRecordID);
+    cy.selectTab(`M_ProductPrice`);
+    cy.selectNthRow(0);
+    cy.openAdvancedEdit();
+    cy.writeIntoStringField('PriceStd', `1`, true, null, true);
```

Comment:
I think/hope we made `writeIntoStringField` behave properly when writing integer. Please try (not sure if it works) using: `cy.writeIntoStringField('PriceStd', 1, true);`

Reply:
It does work.